### DISCUSSION TITLE
HttpKit Server upload handling.

### DISCUSSION THREAD

**Discussion by 0b1kn00b** [babashka/babashka#1757] [Ideas]
Uploads in Clojure tend to be handled by an [Apache library](https://commons.apache.org/proper/commons-fileupload/). 

It was something I noticed was missing when poking about.

**Comment by leobm** [comment]
@0b1kn00b 
I thought I'd have some fun testing whether I could parse the upload stream with a Perl CGI and save it as a file. It was surprisingly easy and even quite fast. I'm sure it would work similarly with Python, etc., but I'm most familiar with Perl. https://github.com/leobm/bb-file-upload

**Reply by borkdude** [reply]
https://github.com/Frozenlock/ring-multipart-nodeps

**Comment by borkdude** [comment]
Check this one out: https://github.com/Frozenlock/ring-multipart-nodeps

**Reply by leobm** [reply]
Okay, that's easier, of course 👍
